Client/Application/QuickActions.swift

```diff
@@ -45,4 +53,42 @@ struct QuickActions {
 
         return true
     }
+
+    mutating func addDynamicApplicationShortcutItemOfType(type: ShortcutType, withUserData userData: [NSObject : AnyObject] = [NSObject : AnyObject](), toApplication application: UIApplication) -> Bool {
+        // add the quick actions version so that it is always in the user info
+        var userInfo = userData
```

Comment:
Don't think you need to make a copy of the dictionary here since dictionaries are passed by value.